### PR TITLE
New ZoomInFadeOut animation added

### LINKED ISSUE #207: New Animation like Moving Letters Effect 15
@saurabhraj042 requested to add a new animated text like [Effect 15](https://tobiasahlin.com/moving-letters/#15)

### PR DESCRIPTION
**Pull request by saurabhraj042**
Issues:
Fixes #207 

Description:
I added a new ZoomInFadeOut animation. 
Updated the Example app and readme.md accordingly

This is how the animation is looking right now
zoomInFadeOut

### CODE REVIEW

**Review by SirusCodes — CHANGES_REQUESTED** [review]
Add a gif

**Comment by awhitford** [comment]
> Add a gif

See #202 for details.  (We realize that we lack guidance on this; hence the issue.)

**Review by awhitford — CHANGES_REQUESTED** [review]
@aagarwal1012 @SirusCodes

Now that I am looking at this closer, it strikes me that this is really nothing more than the _Scale_ animation that simply short-circuits the _Scale Out_.  This reminds me of the adjustment made to the _Rotate_ animation that offers a flag (`rotateOut`) to short-circuit the _Rotate Out_.  I'm thinking a similar approach makes sense.  What do you think?

File: lib/src/zoom_in_fade_out.dart [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  @override
+  Widget animatedBuilder(BuildContext context, Widget? child) {
+    return ScaleTransition(
+      scale: _scaleIn,
+      child: Opacity(opacity: _fadeIn.value, child: textWidget(text)),
+    );
+  }
+}
+
+class ZoomInFadeOutAnimatedTextKit extends AnimatedTextKit {
```
[the remark this answers is not in the document]
Comment (awhitford) [reply lib/src/zoom_in_fade_out.dart:54]: I now see that @aagarwal1012 did confirm that `Kit` classes are no longer required.  See #211.  As a result, I would drop this class.

(Obviously, there is still the grander question of:  "Is this sufficiently different than the existing _Scale_ animation?")

[the remark this answers is not in the document]
Comment (saurabhraj042) [reply lib/src/zoom_in_fade_out.dart:54]: @awhitford Can you please guide me how should I do that , I'm completely clueless here  !

**Comment by SirusCodes** [comment]
> @aagarwal1012 @SirusCodes
> 
> Now that I am looking at this closer, it strikes me that this is really nothing more than the _Scale_ animation that simply short-circuits the _Scale Out_. This reminds me of the adjustment made to the _Rotate_ animation that offers a flag (`rotateOut`) to short-circuit the _Rotate Out_. I'm thinking a similar approach makes sense. What do you think?

I feel it just scales in rather than out. So, we can add an enum `Scale.in` and `Scale.out` to see what animation user is expecting🤔

**Comment by aagarwal1012** [comment]
> Is this sufficiently different than the existing Scale animation?

Yes, this animation is somewhat an upscaled version of the Scale Animation. I would prefer rather than creating a separate `ZoomInFadeOut` animated text -- we should rather add some more customizable parameter to the Scale Animated Text so that it could mimic any animation possibility related to the zoom-in/zoom-out effects possible.

**Comment by saurabhraj042** [comment]
@aagarwal1012 at first I thought to do the same, but because of the mention to create a new effect I did this.So should I go ahead and make changes in the Scale animation itself

**Comment by aagarwal1012** [comment]
@saurabhraj042, before proceeding further, I just wanted to see what @awhitford @SirusCodes react to this idea. Let's wait for their response.

**Comment by awhitford** [comment]
> I would prefer rather than creating a separate `ZoomInFadeOut` animated text -- we should rather add some more customizable parameter to the Scale Animated Text so that it could mimic any animation possibility related to the zoom-in/zoom-out effects possible.

I agree with this sentiment simply because of the similarity of the proposed `ZoomInFadeOut` with `Scale`, it seems more appropriate to tweak the `Scale` -- especially when it literally looks like some simple short-circuiting.

Note that the idea of tweaking an existing Animation to do _too much_ is also a risk.  However, in this case, it seems like the best approach.

I suggest submitting a fresh PR that enhances Scale to also offer Effect 15.

**Comment by saurabhraj042** [comment]
@awhitford @SirusCodes @aagarwal1012  if you all agree with this then please open an issue and assign it to me !

**Comment by awhitford** [comment]
> if you all agree with this then please open an issue and assign it to me !

@saurabhraj042 You can still reference #207.  (The lineage of comments is a helpful record.)

**Comment by SirusCodes** [comment]
> if you all agree with this then please open an issue and assign it to me !

I would like to plan this out in https://github.com/aagarwal1012/Animated-Text-Kit/discussions/212 and then we can continue with the progress

**Comment by saurabhraj042** [comment]
@aagarwal1012 @awhitford @SirusCodes So should I drop the changes made in this PR because as in the Version 3 we are planning to drop the kit class and that will require refactoring all other animation classes also so merging this PR will be of no good.

**Comment by aagarwal1012** [comment]
@saurabhraj042, I would suggest dropping this PR for now and let's start from a fresh piece of code.

**Comment by saurabhraj042** [comment]
@aagarwal1012  how about we change the legacy kit class first ? can you assign that issue to me

**Comment by aagarwal1012** [comment]
@saurabhraj042, #216 have been created by @awhitford that should be fine for now.

**Comment by aagarwal1012** [comment]
Closing this PR, rework required!

**Pull request closed without merging**